DQMOffline/Trigger/plugins/SOSMonitor.cc

```diff
@@ -0,0 +1,316 @@
+  mon.denominator = ibooker.book2D(histname+"_denominator", histtitle+" (denominator)", nbinsX, arrX, nbinsY, arrY);
+}
+
+void SOSMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
+				 edm::Run const        & iRun,
+				 edm::EventSetup const & iSetup) 
+{  
+  
+  std::string histname, histtitle, histtitle2;
+
+  std::string currentFolder = folderName_ ;
+  ibooker.setCurrentFolder(currentFolder.c_str());
+  if (turn_on_=="met") histtitle2="PFMET ";
+  else histtitle2="Pt(mu2)";
+  histname = turn_on_ ; histtitle=histtitle2;
+  bookME(ibooker,sosMON_,histname,histtitle,sos_binning_.nbins,sos_binning_.xmin, sos_binning_.xmax);
+  setTitle(sosMON_,histtitle2+"[GeV]","events / [GeV]");
+
+  histname = turn_on_+"_variable"; histtitle=histtitle2;
+  bookME(ibooker,sosMON_variableBinning_,histname,histtitle,sos_variable_binning_);
+  setTitle(sosMON_variableBinning_,histtitle2+"[GeV]","events / [GeV]");
+
+  histname = turn_on_+"VsLS"; histtitle = histtitle2+"vs LS";
+  bookME(ibooker,sosVsLS_,histname,histtitle,ls_binning_.nbins, ls_binning_.xmin, ls_binning_.xmax,sos_binning_.xmin, sos_binning_.xmax);
+  setTitle(sosVsLS_,"LS",histtitle2+"[GeV]");
+
+   histname = turn_on_+"_Phi"; histtitle = histtitle2+"phi";
+  bookME(ibooker,sosPhiMON_,histname,histtitle, phi_binning_.nbins, phi_binning_.xmin, phi_binning_.xmax);
+  setTitle(sosPhiMON_,histtitle2+"#phi","events / 0.1 rad");
+  
+histname = turn_on_+"_PhiEta"; histtitle = histtitle2+"phiEta";
+  bookME(ibooker,sosPhiEtaMON_,histname,histtitle, phi_binning_.nbins, phi_binning_.xmin, phi_binning_.xmax,eta_binning_.nbins,eta_binning_.xmin,eta_binning_.xmax);
+  setTitle(sosPhiEtaMON_,histtitle2+" #phi",histtitle2+" #eta");
+  
+ histname = turn_on_+"ht" ; histtitle=turn_on_+"ht";
+  bookME(ibooker,sosHTMON_,histname,histtitle,sos_binning_.nbins,sos_binning_.xmin, sos_binning_.xmax);
+  setTitle(sosHTMON_,"HT [GeV]","events / [GeV]");
+
+
+  // Initialize the GenericTriggerEventFlag
+  if ( num_genTriggerEventFlag_ && num_genTriggerEventFlag_->on() ) num_genTriggerEventFlag_->initRun( iRun, iSetup );
+  if ( den_genTriggerEventFlag_ && den_genTriggerEventFlag_->on() ) den_genTriggerEventFlag_->initRun( iRun, iSetup );
+
+}
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+void SOSMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup)  {
+
+  // Filter out events if Trigger Filtering is requested
+  if (den_genTriggerEventFlag_->on() && ! den_genTriggerEventFlag_->accept( iEvent, iSetup) ) return;
+
+  edm::Handle<reco::PFMETCollection> metHandle;
+  iEvent.getByToken( metToken_, metHandle );
+  reco::PFMET pfmet = metHandle->front();
+  if ( ! metSelection_( pfmet ) ) return;
+  
+  float met = pfmet.pt();
+  float phi = pfmet.phi();
+  int jet60=0;
```

Comment:
ok, I see that all comments I made in your branch have not been addressed yet
please, commits the needed changes
then, I'll continue the review